### PR TITLE
Move paper, books and book shelves into separate mod

### PR DESCRIPTION
**Pull request by rubenwardy**

### CODE REVIEW

File: mods/books/init.lua [hunk 1]
```diff
@@ -0,0 +1,89 @@
+books = {}
+
+dofile(minetest.get_modpath("books") .. "/craftitems.lua")
+dofile(minetest.get_modpath("books") .. "/nodes.lua")
+dofile(minetest.get_modpath("books") .. "/crafting.lua")
+
+if mg_name == "v6" then
```
Comment (paramat) [inline mods/books/init.lua]: You need to 'get mapgen setting' first https://github.com/minetest/minetest_game/blob/master/mods/default/mapgen.lua#L1471

**Comment by paramat** [comment]
I'm not sure if papyrus and it's mapgen are a part of books mod, they could perhaps be seen as a desirable part of standard mapgen even if not used for paper.

**Comment by everamzah** [comment]
I don't think papyrus and paper belong in `books`, anymore than sand and glass belong in `vessels`.

**Comment by rubenwardy** [comment]
Sand is a part of the environment and is common in real life, it's used to make up beaches. Glass is a building material which can also be made to make vessels.

Does paper make sense without books or bookshelves?

Papyrus is the source of paper, like clay block -> clay.
It has no reason to exist without paper.

Papyrus being used for sugar is just stupid, it's like making chocolate from dirt.

**Comment by everamzah** [comment]
It doesn't really matter, to me, but I think you might be onto something.

I wanted to illustrate only that `vessels` mod contains only the shelf node, and glass and steel vessel craftitems.

On the other hand, why not modularize minetest_game where you can?

**Comment by everamzah** [comment]
>It has no reason to exist without paper.
I disagree.  Spotting papyrus near water among lily pads is a real treat.  Farming them for decorative purposes is also a plus.

**Comment by Fixer-007** [comment]
Papyrus is essential part for mapgen.

**Comment by rubenwardy** [comment]
> Papyrus is essential part for mapgen.

How is it? In a world without paper, it makes little sense for it to exist

**Comment by paramat** [comment]
It's only used for paper in MTGame, but then, the raw materials for a mod's items are not usually moved into each mod. Still unsure about this though and still neutral on this.

**Comment by everamzah** [comment]
Now that I think about it, papyrus might be better suited for `flowers` since it grows.

**Comment by Fixer-007** [comment]
It is part of some biomes, obviously, subgame makers can disable it anyway..

**Comment by rubenwardy** [comment]
It appears that I've misread and papyrus spawns in swamp mapgens. I'm guessing that it looks like a type of reed there? If without paper, it's just a reed, then it's fine

> Papyrus could be used for other purposes other than paper.

Some lazy mod makers use it for sugar, but that's a bad use imo.

**Comment by paramat** [comment]
Yes it's an important feature of mapgen, so i think papyrus node and it's generation should stay in default. I don't oppose the rest being moved to a mod.

**Comment by rubenwardy** [comment]
Texture change will break:

* [castle](https://github.com/mimilus/castle)
* [carpets](https://github.com/Rui914/carpets)

Dependency change may break:

* homedecor_3d_extras
* [carpets](https://github.com/Rui914/carpets)

**Comment by paramat** [comment]
Breakage seems minor and we can make a post on the forum.

**Comment by t4im** [comment]
Maybe sort the file content by function
* books.lua
* shelf.lua
* papyrus.lua (if it stays)

That way you'd have things together when someone works on it, instead of having to go through all of crafts.lua, craftitems.lua, functions.lua, aliases.lua, abms.lua, nodes.lua, code-comments.lua all the time to collect related things.

**Comment by DonBatman** [comment]
@t4im I like that idea. I am working on a game and I find it a pain to find the right things.

**Comment by rubenwardy** [comment]
moved papyrus back to default. Now only paper, books and book shelves have been moved

**Comment by paramat** [comment]
:+1: If this is merged please can you then make an announcement in the news section of the forum, and maybe also leave a link to that announcement in the broken mod threads?
@sofar what do you think about this?

**Comment by paramat** [comment]
This apparently makes subgames easier to work on, which is something we certainly want to encuorage. Seeing as just a few mods will need to make very minor changes what are the arguments against? I was very much against the previous attempt to split default into many small mods, and still am, this however seems restrained and only slightly disruptive.

**Comment by VanessaE** [comment]
:-1: this is the second or third change in a trend that I frankly do not like.  What possible benefit is there to players by making these "modular" changes?  If you're a subgame author, it's already trivial to override/disable sections of the standard game content.

You're breaking mods for no good reason.

**Comment by paramat** [comment]
It might be the last of these changes, from what the author has written on IRC.

**Comment by Ferk** [comment]
I think the biggest benefit of this for a subgame author, other than being easier to locate things, is being able to keep up with the changes done in MTG for those things borrowed.

If you override default (which you should as it's the whole point of making a subgame and not a modpack) you filled it with conflicting changes but you still want to get the latest fancy new book formspec from MTG, then you just need to copy over the book mod folder to update it. instead of hunting to cherry-pick the changes in the default mod code.

**Comment by paramat** [comment]
I'm considering merging this as the author is a possible future dev, and because all other mtgame devs have disappeared for weeks.

**Comment by Ekdohibs** [comment]
This is too likely to break mods: any mod that will check ```stack:get_name() == "default:book"``` will break for instance. So, :-1: from me.

File: mods/books/init.lua [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+dofile(minetest.get_modpath("books") .. "/craftitems.lua")
+dofile(minetest.get_modpath("books") .. "/nodes.lua")
+dofile(minetest.get_modpath("books") .. "/crafting.lua")
+
+minetest.register_alias("paper", "books:paper")
+minetest.register_alias("book", "books:book")
+minetest.register_alias("bookshelf", "books:bookshelf")
+minetest.register_alias("default:paper", "books:paper")
+minetest.register_alias("default:default:book", "books:book")
```
Comment (SmallJoker) [inline mods/books/init.lua]: Can't be more default than default.

**Comment by paramat** [comment]
I'm unsure about this now, fairly neutral so removed my +1/

**Comment by rubenwardy** [comment]
That's quite a big problem recognised by nore. It's not that bad if you only need to add a line to the dependencies, but having to edit code and add backwards compatibility for older minetest versions is too much, and outweighs the gain.

Whilst moving just books etc out has little effect, the added effect of moving almost everything non mapgens out of default would have made dependency management easier when running a game which isn't MTG+mods, but a differing subgame (like rpgtest). It should have been done in the 0.4 alpha days though. I guess a better way to do this may be to have item dependencies, so a mod can say what nodes it needs to exists to work properly rather than fail silently. For now I'll just leave it and work on other things

**Comment by paramat** [comment]
Could still move these to a new file within default?

**Comment by VanessaE** [comment]
I don't see any issue with doing that, sure.

**Pull request closed without merging**